### PR TITLE
chore: Update README for install instructions

### LINKED ISSUE #24: Update documentation / `install.yaml` to use `initContainer` patch instead of Extension CR

### LINKED ISSUE #62: Update since argocd-extensions is deprecated
Now that `argocd-extensions` is [deprecated](https://github.com/argoproj-labs/argocd-extensions?tab=readme-ov-file#deprecation-notice), will this repo be updated for the new way of defining extensions?

### LINKED ISSUE #68: Revamping README.md based on ArgoCD Extension Installer
The goal is to replace information related to an outdated mechanism with details about the new and enhanced method based on [Argo CD Extension Installer](https://github.com/argoproj-labs/argocd-extension-installer)

### PR DESCRIPTION
**Pull request by standardloop**
Closes #68, Closes #24, Closes #62

Open to making this more verbose, let me know.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,13 +5,43 @@ The project introduces the Argo Rollout dashboard into the Argo CD Web UI.
...
-# Quick Start
+## Install UI extension
 
-- Install Argo CD and Argo CD Extensions Controller: https://github.com/argoproj-labs/argocd-extensions
-- Create `argo-rollouts` extension in `argocd` namespace
+The UI extension needs to be installed by mounting the React component
+in Argo CD API server. This process can be automated by using the
+[argocd-extension-installer][1]. This installation method will run an
+init container that will download, extract and place the file in the
+correct location.
```
Comment (ashutosh16) [inline README.md:14]: This approach uses the [argocd-extension-installer](https://github.com/argoproj-labs/argocd-extension-installer) init container which runs during the startup of the argocd server.  
The init container downloads and extracts the JS file to `/tmp/extensions`. The argocd interface mounts the external JS file within the rollout resource.

Comment (standardloop) [reply README.md]: Updated, please take a look and re-review at your leisure.

**Comment by standardloop** [comment]
> Make sense to include also instructions about how to install it using other alternatives as ArgoCD Helm chart?

@jsolana  Updated to have helm options as well.

**Comment by standardloop** [comment]
> This is a nice improvement and I used it today. Any chance it can get merged soon?

Happy to hear you found some use.

Sadly I don't have write access to this respository

**Comment by standardloop** [comment]
@rbreeze able to review? I see you have had recent contributions to this repo

**Comment by taer** [comment]
This is AMAZING!  Thanks for the EZ button. :)

**Comment by rgarrigue** [comment]
Wish this was merged before I looked all over the repos for how to

**Comment by standardloop** [comment]
@ashutosh16 @alexmt @rbreeze @alexef 

Can we get this merged in?

**Review by rbreeze — APPROVED** [review]

**Pull request merged**